### PR TITLE
Add status about registry background task to /meta.

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

**Comment by fredrikekre** [comment]
Perhaps we can try to self-heal?
````diff
$ git diff
diff --git a/src/PkgServer.jl b/src/PkgServer.jl
index 4f0029c..d2334a0 100644
--- a/src/PkgServer.jl
+++ b/src/PkgServer.jl
@@ -84,13 +84,23 @@ function start(;kwargs...)
 
     @sync begin
         global last_registry_update
+        function start_registry_updater()
+            t = @spawn while true
+                last_registry_update = now()
+                sleep(1)
+                @try_printerror begin
+                    forget_failures()
+                    update_registries()
+                end
+            end
+            return t
+        end
         @spawn while true
-            last_registry_update = now()
-            sleep(1)
-            @try_printerror begin
-                forget_failures()
-                update_registries()
+            t = start_registry_updater()
+            if istaskstarted(t) && (istaskdone(t) || istaskfailed(t))
+                t = start_registry_updater()
             end
+            sleep(10)
         end
         @info("server listening", config.listen_addr)
         HTTP.listen(config.listen_addr.host, config.listen_addr.port) do http
````

File: src/meta.jl [hunk 1]
```diff
@@ -140,6 +140,7 @@ function serve_meta(http::HTTP.Stream)
         "julia_version" => string(VERSION),
         "start_time" => string(time_start),
         "last_registry_update" => string(last_registry_update),
+        "registry_update_task_running" => string(istaskstarted(registry_task) & !istaskdone(registry_task)),
```
Comment (StefanKarpinski) [inline src/meta.jl]: Might be more informative to show what exact state the task is in, i.e.
```jl
task_state =
    istaskdone(registry_task)    ? "done"    :
    istaskfailed(registry_task)  ? "failed"  :
    istaskstarted(registry_task) ? "started" : "not started"
```
Then the entry is:
```jl
"registry_update_task" => task_state,
```

Comment (fredrikekre) [reply src/meta.jl]: Sure. A failed task is both "done" and "failed", so perhaps reverse the order and prioritize "failed" over "done"?

Comment (StefanKarpinski) [reply src/meta.jl]: Right. I had assumed the task states were mutually exclusive, but of course they're not.

**Comment by staticfloat** [comment]
I'd rather restart the whole process than restart that one task.  It may also be useful for us to serve a string of the exception and backtrace of the task, for debugging purposes.

**Pull request merged**